packages/service-library/src/servicelib/rabbitmq/_utils.py

```diff
@@ -70,4 +70,18 @@ async def declare_queue(
     if not exclusive_queue:
         # NOTE: setting a name will ensure multiple instance will take their data here
         queue_parameters |= {"name": exchange_name}
+
+    # NOTE: if below line raises something similar to ``ChannelPreconditionFailed: PRECONDITION_FAILED``
+    # most likely someone changed the signature of the queues (parameters etc...)
+    # Safest way to deal with it:
+    #   1. check whether there are any messages for the existing queue in rabbitmq
+    #   2. NO messages -> delete queue
+    #   3. Found messages:
+    #        - save messages
+    #        - delete queue
+    #        - restore messages
+    # Why is this the safest, with an example?
+    #   1. a user bought 1000$ of credits
+    #   2. for some reason resource usage tracker is unavailable and the messages is stuck in the queue
+    #   3. if the queue is deleted, the action relative to this transaction will be lost
```

Comment:
what about catching the error and giving a meaningful error in the logs? we know that rabbit queue parameters should not be changed otherwise it fails.

Reply:
Since I'm not 100% sure what error type will be raised. I left it in the comments, in case something else which looks similar pops up.

Reply:
i tend to agree with @sanderegg . Better in the logger.exception than in the documentation.

Reply:
I guess is `aiormq.exceptions.ChannelPreconditionFailed`

```
aiormq.exceptions.ChannelPreconditionFailed: PRECONDITION_FAILED - inequivalent arg 'x-dead-letter-exchange' for queue 'io.simcore.service.wallets-credit-limit-reached' in vhost '/': received the value 'delayed_io.simcore.service.wallets-credit-limit-reached' of type 'longstr' but current is none`

Reply:
@pcrespov what happens if a different execution is raised? I guess we'll have a traceback right? that would also work

Reply:
@pcrespov please have a look if it's better like this. If yes, please resolve the thread and allow it to merge.

Reply:
ok :)